feat: new UDFs for set-like operations on Arrays (#5548)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/scalar-functions.md

Accepts any `ARRAY` type. The type of the second param must match the element type of the `ARRAY`.

### ``ARRAY_DISTINCT``

```sql
ARRAY_DISTINCT([1, 2, 3])
```

Returns an array of all the distinct values, including NULL if present, from the input array.

> The output array elements will be in order of their first occurrence in the input.

Reviewer: ```suggestion
The output array elements are in order of their first occurrence in the input.
```

Returns NULL if the input array is NULL.

Examples:
```sql 
ARRAY_DISTINCT(ARRAY[1, 1, 2, 3, 1, 2])  => [1, 2, 3]
ARRAY_DISTINCT(ARRAY['apple', 'apple', NULL, 'cherry'])  => ['apple', NULL, 'cherry']
```

### ``ARRAY_EXCEPT``

```sql
ARRAY_EXCEPT(array1, array2)
```

> Returns an array of all the distinct elements from an array except for those also present in a second array. The order of entries in the first array is preserved although any duplicates are removed. 

Reviewer: ```suggestion
Returns an array of all the distinct elements from an array, except for those also present in a second array. The order of entries in the first array is preserved, but duplicates are removed. 
```
